### PR TITLE
feat: http proxy config for httpAudioSourceManager

### PR DESCRIPTION
**Pull request by ToxicMushroom**
Useful for stopping bad actors from IP-grabbing your main music server and attacking it.
Tested with squidproxy

### CODE REVIEW

**Comment by Walkyst** [comment]
How it supposed to connect if proxy behind some basic auth like login and password?

**Comment by ToxicMushroom** [comment]
> How it supposed to connect if proxy behind some basic auth like login and password?

I've added an option for authentication, I don't have a setup to test this though

File: LavalinkServer/application.yml.example [hunk 1]
```diff
@@ -28,6 +28,9 @@ lavalink:
     #youtubeConfig: # Required for avoiding age restrictions by YouTube. Guide https://github.com/Walkyst/lavaplayer-fork/issues/18
       #PAPISID:      ""
       #PSID:         ""
+    #httpConfig:
+      #proxyHost: "localhost"
+      #proxyPort: 3128
```
Comment (freyacodes) [inline LavalinkServer/application.yml.example]: This needs to be documented with comments, and this is missing username and password

**Review by freyacodes — CHANGES_REQUESTED** [review]

**Comment by freyacodes** [comment]
It still needs a description of why you would ever want to set up a proxy

**Review by freyacodes — APPROVED** [review]

**Pull request merged**